sktime/alignment/dtw_python.py

```diff
@@ -0,0 +1,337 @@
+            "none" (default) - no windowing
+            "sakoechiba" - a band around main diagonal
+            "slantedband" - a band around slanted diagonal
+            "itakura" - Itakura parallelogram
+    open_begin, open_end: boolean; whether to perform open-ended alignments
+        open_begin = whether alignment open ended at start (low index)
+        open_end = whether alignment open ended at end (high index)
+    variable_to_align: string; which variable to use for univariate alignment
+        default: first variable in X as passed to fit
+    """
+
+    _tags = {
+        "capability:multiple-alignment": False,  # can align more than two sequences?
+        "capability:distance": True,  # does compute/return overall distance?
+        "capability:distance-matrix": True,  # does compute/return distance matrix?
+    }
+
+    def __init__(
+        self,
+        dist_method="euclidean",
+        step_pattern="symmetric2",
+        window_type="none",
+        open_begin=False,
+        open_end=False,
+        variable_to_align=None,
+    ):
+        """Construct instance."""
+        super(AlignerDTWdist, self).__init__()
+
+        self.dist_method = dist_method
+        self.step_pattern = step_pattern
+        self.window_type = window_type
+        self.open_begin = open_begin
+        self.open_end = open_end
+        self.variable_to_align = variable_to_align
+
+    def _fit(self, X, Z=None):
+        """Fit alignment given series/sequences to align.
+
+            core logic
+
+        Parameters
+        ----------
+        X: list of pd.DataFrame (sequence) of length n - panel of series to align
+        Z: pd.DataFrame with n rows, optional; metadata, row correspond to indices of X
+
+        Writes to self
+        --------------
+        alignment - computed alignment from dtw package
+        X - reference to input X
+        """
+        # soft dependency import of dtw
+        from dtw import dtw
+
+        # these variables from self are accessed
+        dist_method = self.dist_method
+        step_pattern = self.step_pattern
+        window_type = self.window_type
+        open_begin = self.open_begin
+        open_end = self.open_end
+
+        # shorthands for 1st and 2nd sequence
+        X1 = X[0]
+        X2 = X[1]
+
+        # retrieve column to align from data frames, convert to np.array
+        var_to_align = self.variable_to_align
+        if var_to_align is None:
+            var_to_align = X1.columns.values[0]
+
+        if var_to_align not in X1.columns.values:
+            raise ValueError(
+                f"X[0] does not have variable {var_to_align} used for alignment"
+            )
+        if var_to_align not in X2.columns.values:
+            raise ValueError(
+                f"X[1] does not have variable {var_to_align} used for alignment"
+            )
+
+        X1vec = X1[var_to_align].values
+        X2vec = X2[var_to_align].values
```

Comment:
instead of `vec`, it would be clearer to say what the machine type is. so either `X1_np` or `X1_list` depending on if it is numpy array or python list.